include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -1100,14 +1097,19 @@ class alignment_file_output_format<format_sam>
                       !std::Integral<std::remove_reference_t<ref_id_type>> &&
                       !detail::is_type_specialisation_of_v<std::remove_reference_t<ref_id_type>, std::optional>)
         {
-            static_assert(ImplicitlyConvertibleTo<ref_id_type &, typename decltype(header.ref_dict)::key_type>,
-                      "The ref_id type is not convertible to the reference id information stored in the "
-                      "reference dictionary of the header object.");
 
             if (options.sam_require_header && !std::ranges::empty(ref_id))
             {
-                if ((header.ref_dict).count(ref_id) == 0) // no reference id matched
-                    throw format_error{std::string("The ref_id '") + std::string(ref_id) +
+                using header_ref_id_type = std::remove_reference_t<decltype(header.ref_ids()[0])>;
+
+                static_assert(ImplicitlyConvertibleTo<value_type_t<ref_id_type>, value_type_t<header_ref_id_type>>,
+                              "The ref_id type is not convertible to the reference id information stored in the "
+                              "reference dictionary of the header object.");
+
+                auto key = ref_id | std::ranges::to<header_ref_id_type>;
+
+                if ((header.ref_dict).count(key) == 0) // no reference id matched
+                    throw format_error{std::string("The ref_id '") + std::string(key) +
```

Comment:
```suggestion
                    throw format_error{std::string_view("The ref_id '") + std::string_view(key) +
```

Reply:
Well I can not concatenate string views. I changed the whole by using the `detail::to_string` function